provider/bidengine/pricing.go

```diff
@@ -66,3 +174,105 @@ func calculatePriceRange(gspec *dtypes.GroupSpec) (sdk.Coin, sdk.Coin) {
 
 	return sdk.NewCoin(rmax.Denom, cmin), sdk.NewCoin(rmax.Denom, cmax)
 }
+
+type shellScriptPricing struct {
+	path         string
+	processLimit chan int
+}
+
+func MakeShellScriptPricing(path string, processLimit uint) (BidPricingStrategy, error) {
+	if len(path) == 0 {
+		return nil, errors.New("script path cannot be the empty string")
+	}
+	if processLimit == 0 {
+		return nil, fmt.Errorf("process limit must be greater than zero")
+	}
+
+	result := shellScriptPricing{
+		path:         path,
+		processLimit: make(chan int, processLimit),
+	}
+
+	// Use the channel as a semaphore to limit the number of processes created for computing bid processes
+	// Most platforms put a limit on the number of processes a user can open. Even if the limit is high
+	// it isn't a good idea to open thuosands of processes.
```

Comment:
I like the defensive programming, but I just don't know if it's necessary at this point.

Reply:
Is there any limit to the number of orders that a provider can receive in a given time period? Without this check it feels like an attack vector against providers.

Reply:
It's limited by number of orders that can make it into a block - I'd be surprised to see 500 in a block (~6 seconds).

I'm kinda neutral on it.  It feels like it should be more baked but it's okay to leave for now if you feel strongly about it.  It is a good thing to protect after all.

In a follow-up issue it'd be nice to

1. make it cancel-able (exits early if `ctx.Done()` is readable.
1. set a limit on execution time (there's a `cmd.RunContext()` I believe).

Reply:
This is done now.